src/Microsoft.AspNet.Razor.Runtime/TagHelpers/TagHelperOutput.cs

```diff
@@ -138,12 +151,12 @@ public string GenerateStartTag()
         /// <summary>
         /// Generates the <see cref="TagHelperOutput"/>'s <see cref="PreContent"/>.
         /// </summary>
-        /// <returns><c>string.Empty</c> if <see cref="SelfClosing"/> is <c>true</c>. <see cref="PreContent"/> 
-        /// otherwise.
+        /// <returns><c>string.Empty</c> if <see cref="TagName"/> is <c>null</c> or whitespaces.
+        /// Otherwise, <see cref="PreContent"/>.
```

Comment:
incorrect.  should say
`<c>string.Empty</c> if <see cref="GenerateStartTag"/> will return a non-empty `string` and <see cref="SelfClosing"/> is <c>true</c>`
or perhaps
`<c>string.Empty</c> if <see cref="TagName"/> is not <c>null</c> or whitespace and <see cref="SelfClosing"/> is <c>true</c>`.